internal/jimm/juju/model.go

```diff
@@ -435,14 +444,47 @@ func (j *JujuManager) ModelStatus(ctx context.Context, user *openfga.User, mt na
 	err := j.doModelAdmin(ctx, user, mt, func(m *dbmodel.Model, api API) error {
 		ms.OwnerTag = m.Owner.Tag().String()
 		ms.ModelTag = mt.String()
-		return api.ModelStatus(ctx, &ms)
+		err := api.ModelStatus(ctx, &ms)
+		if err != nil {
+			// If the model is not found on the backing controller then
+			// we delete the model from JIMM.
+			if errors.ErrorCode(err) == errors.CodeNotFound {
+				errDelete := j.deleteModel(ctx, mt)
+				if errDelete != nil {
+					return errDelete
+				}
+			}
+			return err
+		}
+		return nil
 	})
 	if err != nil {
 		return nil, errors.E(op, err)
 	}
 	return &ms, nil
 }
 
+// deleteModel deletes the model with the given ModelTag from JIMM's database and
+// clean the openfga tuples related to the model.
+func (j *JujuManager) deleteModel(ctx context.Context, mt names.ModelTag) error {
+	err := j.Database.DeleteModel(ctx, &dbmodel.Model{
+		UUID: sql.NullString{
+			String: mt.Id(),
+			Valid:  true,
+		},
+	})
+	if err != nil {
+		zapctx.Error(ctx, "failed to delete model from database", zap.String("model", mt.Id()), zap.Error(err))
+		return err
+	}
+	err = j.permissionManager.RemoveModel(ctx, mt)
```

Comment:
I like the use of the permission manager. It also would've been possible to just do `j.authSvc.RemoveModel()` in here right? Maybe one day we'll remove the auth service from the Juju manager. Edit: I've added some more thoughts on this in another comment.